test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.cs

```diff
@@ -317,6 +317,63 @@ public static TheoryData<Saml2TheoryData> RoundTripTokenTheoryData
                         ValidAudience = Default.Audience,
                         ValidIssuer = Default.Issuer,
                     }
+                },
+                    new Saml2TheoryData
+                {
+                    First = true,
+                    TestId = nameof(Default.ClaimsIdentity) + nameof(KeyingMaterial.RSASigningCreds_2048),
+                    TokenDescriptor = new SecurityTokenDescriptor
+                    {
+                        Expires = DateTime.UtcNow + TimeSpan.FromDays(1),
+                        Audience = Default.Audience,
+                        SigningCredentials = KeyingMaterial.RSASigningCreds_2048,
+                        Issuer = Default.Issuer,
+                        Subject = Default.ClaimsIdentity
+                    },
+                    ValidationParameters = new TokenValidationParameters
+                    {
+                        IssuerSigningKey = KeyingMaterial.RSASigningCreds_2048_Public.Key,
+                        ValidAudience = Default.Audience,
+                        ValidIssuer = Default.Issuer,
+                    },
+                },
+                new Saml2TheoryData
+                {
+                    First = true,
+                    TestId = nameof(Default.ClaimsIdentity) + nameof(KeyingMaterial.RSASigningCreds_2048_FromRsa),
+                    TokenDescriptor = new SecurityTokenDescriptor
+                    {
+                        Expires = DateTime.UtcNow + TimeSpan.FromDays(1),
+                        Audience = Default.Audience,
+                        SigningCredentials = KeyingMaterial.RSASigningCreds_2048_FromRsa,
+                        Issuer = Default.Issuer,
+                        Subject = Default.ClaimsIdentity
+                    },
+                    ValidationParameters = new TokenValidationParameters
+                    {
+                        IssuerSigningKey = KeyingMaterial.RSASigningCreds_2048_FromRsa_Public.Key,
+                        ValidAudience = Default.Audience,
+                        ValidIssuer = Default.Issuer,
+                    },
+                },
+                new Saml2TheoryData
+                {
+                    First = true,
+                    TestId = nameof(Default.ClaimsIdentity) + nameof(KeyingMaterial.JsonWebKeyRsa256SigningCredentials),
+                    TokenDescriptor = new SecurityTokenDescriptor
+                    {
+                        Expires = DateTime.UtcNow + TimeSpan.FromDays(1),
+                        Audience = Default.Audience,
+                        SigningCredentials = KeyingMaterial.JsonWebKeyRsa256SigningCredentials,
```

Comment:
If you can keep things in order, it is easier to find. It's natural to get out of order but a, b, c is easier to see follow.